packages/firebase_ml_vision/example/lib/detector_painters.dart

```diff
@@ -115,55 +106,103 @@ class LabelDetectorPainter extends CustomPainter {
+  VisionEdgeLabelDetectorPainter(this.imageSize, this.labels);
+
+  final Size imageSize;
+  final List<VisionEdgeImageLabel> labels;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    final ui.ParagraphBuilder builder = ui.ParagraphBuilder(
+      ui.ParagraphStyle(
+          textAlign: TextAlign.left,
+          fontSize: 23.0,
+          textDirection: TextDirection.ltr),
+    );
+
+    builder.pushStyle(ui.TextStyle(color: Colors.green));
+    for (VisionEdgeImageLabel label in labels) {
+      builder.addText('Label: ${label.text}, '
+          'Confidence: ${label.confidence.toStringAsFixed(2)}\n');
+    }
+    builder.pop();
+
+    canvas.drawParagraph(
+      builder.build()
+        ..layout(ui.ParagraphConstraints(
+          width: size.width,
+        )),
+      const Offset(0.0, 0.0),
+    );
+  }
+
+  @override
+  bool shouldRepaint(VisionEdgeLabelDetectorPainter oldDelegate) {
+    return oldDelegate.imageSize != imageSize || oldDelegate.labels != labels;
   }
 }
 
 // Paints rectangles around all the text in the image.
 class TextDetectorPainter extends CustomPainter {
-  TextDetectorPainter(this.absoluteImageSize, this.visionText);
+  TextDetectorPainter(this.imageSize, this.visionText);
 
-  final Size absoluteImageSize;
+  final Size imageSize;
   final VisionText visionText;
 
   @override
   void paint(Canvas canvas, Size size) {
-    final double scaleX = size.width / absoluteImageSize.width;
-    final double scaleY = size.height / absoluteImageSize.height;
-
-    Rect scaleRect(TextContainer container) {
-      return Rect.fromLTRB(
-        container.boundingBox.left * scaleX,
-        container.boundingBox.top * scaleY,
-        container.boundingBox.right * scaleX,
-        container.boundingBox.bottom * scaleY,
-      );
-    }
-
     final Paint paint = Paint()
       ..style = PaintingStyle.stroke
       ..strokeWidth = 2.0;
 
+    Rect _getRect(TextContainer container) {
+      return _scaleRect(
+        rect: container.boundingBox,
+        imageSize: imageSize,
+        widgetSize: size,
+      );
+    }
+
     for (TextBlock block in visionText.blocks) {
       for (TextLine line in block.lines) {
         for (TextElement element in line.elements) {
           paint.color = Colors.green;
-          canvas.drawRect(scaleRect(element), paint);
+          canvas.drawRect(_getRect(element), paint);
         }
 
         paint.color = Colors.yellow;
-        canvas.drawRect(scaleRect(line), paint);
+        canvas.drawRect(_getRect(line), paint);
       }
 
       paint.color = Colors.red;
-      canvas.drawRect(scaleRect(block), paint);
+      canvas.drawRect(_getRect(block), paint);
     }
   }
 
   @override
   bool shouldRepaint(TextDetectorPainter oldDelegate) {
-    return oldDelegate.absoluteImageSize != absoluteImageSize ||
+    return oldDelegate.imageSize != imageSize ||
         oldDelegate.visionText != visionText;
   }
 }
+
+Rect _scaleRect({
```

Comment:
Was this copied from the [ML Kit demo](https://github.com/bparrishMines/mlkit_demo). I recently archived that and the most updated code is inside this plugin. You should be able to uncheck this.